### PR TITLE
Ui 736 search

### PR DESCRIPTION
**Pull request by sheela-svmx**
- A search component that renders a input and icon and being able to handle onChange, onSubmit
- Storybook is not exclusive with stories. If you have ideas please suggest.

### CODE REVIEW

File: lib/components/Search/Search.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+class Search extends Component {
+  handleChange = e => {
+    e.persist();
+    const { setTimeOut, onChange } = this.props;
+    setTimeout(() => {
+      if (onChange) {
+        onChange(e);
+      }
+    }, setTimeOut);
```
Comment (mallik-svmx) [inline lib/components/Search/Search.js]: 'setTimeOut' prop confusing 'DELAY' would be right name in here

File: lib/components/Search/Search.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      className,
+      iconAlign = 'right',
+      placeholder,
+      ...restProps
+    } = this.props;
+
+    delete restProps.setTimeOut;
+    delete restProps.onComplete;
+
+    const inputClassName = classnames(
```
Comment (mallik-svmx) [inline lib/components/Search/Search.js]: searchClassNames?

File: lib/components/Search/Search.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      <div className={inputClassName}>
+        <Input
+          onChange={this.handleChange}
+          onKeyDown={this.handleInputKeyDown}
+          {...restProps}
+        />
+        <span
+          tabIndex={-1}
+          role="button"
+          onClick={this.handleLookupIconClick}
```
Comment (mallik-svmx) [inline lib/components/Search/Search.js]: I dont see handleLookupIconClick and handleInputBlur in this class, where are they coming from?

File: lib/components/Search/Search.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+const defaultProps = {
+  className: '',
+  onComplete: null,
+  placeholder: '',
+  setTimeOut: null,
+};
+
+class Search extends Component {
```
Comment (mallik-svmx) [inline lib/components/Search/Search.js]: Probably, It could just be functional component.

File: lib/components/Search/index.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import Search from './Search';
```
Comment (mallik-svmx) [inline lib/components/Search/index.js]: Its fine, if you would like to have it in single line then: export { default } from './Serach'; and it still works the same way.

File: lib/components/Search/Search.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    if (onKeyDown) {
+      onKeyDown(e);
+    }
+  };
+
+  render() {
+    const {
+      className,
+      iconAlign = 'right',
+      placeholder,
```
Comment (hanzge) [inline lib/components/Search/Search.js]: `placeholder` has never been used here. You can remove it and let it be in the `...restProps`

File: lib/components/Search/Search.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+          onChange={this.handleChange}
+          onKeyDown={this.handleInputKeyDown}
+          {...restProps}
+        />
+        <span
+          tabIndex={-1}
+          role="button"
+          onClick={this.handleLookupIconClick}
+          onBlur={this.handleInputBlur}
+        >
```
Comment (hanzge) [inline lib/components/Search/Search.js]: If it's a clickable `span`, we need to add `cursor: pointer` to css. Or maybe just change it to `<Button type="icon">`?

File: lib/components/Search/Search.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  placeholder: PropTypes.string,
+  /**
+   * Set time out to start the onchange after a certain time
+   */
+  setTimeOut: PropTypes.number,
+};
+
+const defaultProps = {
+  className: '',
+  onComplete: null,
```
Comment (hanzge) [inline lib/components/Search/Search.js]: `onComplete` is not in `propTypes` and never used.

File: lib/components/Search/index.js (hunk 5 above)
Comment (sheela-svmx) [reply lib/components/Search/index.js]: Thanks for pointing it out, I will change it

File: lib/components/Search/Search.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      }
+    }
+    if (onKeyDown) {
+      onKeyDown(e);
+    }
+  };
+
+  handleInputBlur = e => {
+    e.persist();
+    setTimeout(() => {
```
Comment (joshsylvester) [inline lib/components/Search/Search.js]: again here - move the if statements outside of the `setTimeout`, unless the isFocusedInComponent prevents you from doing so - even then at least move the `if (this.props.onBlur) ` outside.

File: stories/SearchStories.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    const oldData = this.state.data;
+    const data = oldData.filter(item => item.name
+      .toLowerCase()
+      .indexOf(e.target.value.toLowerCase()) !== -1);
+    this.setState({ data });
+  };
+
+  render() {
+    return (<div>
+      <Search setTimeOut={1000} onKeyDown={this.onSearch} />
```
Comment (joshsylvester) [inline stories/SearchStories.js]: your demo still refers to `setTimeOut`, not `delay`

in addition - the two params you are passing don't work with each other.  `onKeyDown` doesn't respect the `delay` (should it?) based on the code in `Search.js`

File: lib/components/Search/Search.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+   */
+  onChange: PropTypes.func,
+  /**
+   * Function callback onKeyDown
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * Function callback onSubmit
+   */
+  onSubmit: PropTypes.func,
```
Comment (hanzge) [inline lib/components/Search/Search.js]: In my editor, there's an eslint error on this line - `no-trailing-spaces`. Do you have your eslint on? It should report an error.

Comment (sheela-svmx) [reply lib/components/Search/Search.js]: yes i had missed that when i did my PR comments

**Pull request merged**